File: flutter/packages/measure_flutter/lib/attribute_value.dart
/// Represents a value of an attribute. It can be a string, boolean, integer,
/// or double.
sealed class AttributeValue {
  final Object value;

  const AttributeValue._(this.value);
}

final class StringAttr extends AttributeValue {
  StringAttr(String super.value) : super._();
}

final class BooleanAttr extends AttributeValue {
  BooleanAttr(bool super.value) : super._();
}

final class IntAttr extends AttributeValue {
  IntAttr(int super.value) : super._();
}

final class DoubleAttr extends AttributeValue {
  DoubleAttr(double super.value) : super._();
}

File: kmp/measure-kmp/src/commonMain/kotlin/sh/measure/kmp/attributes/AttributeValue.kt
package sh.measure.kmp.attributes

import kotlin.jvm.JvmInline

/**
 * Represents a value of an attribute. It can be a string, boolean, integer, long, float, or double.
 */
sealed interface AttributeValue {
    val value: Any
}

@JvmInline
value class StringAttr(override val value: String) : AttributeValue

@JvmInline
value class BooleanAttr(override val value: Boolean) : AttributeValue

@JvmInline
value class IntAttr(override val value: Int) : AttributeValue

@JvmInline
value class LongAttr(override val value: Long) : AttributeValue

@JvmInline
value class FloatAttr(override val value: Float) : AttributeValue

@JvmInline
value class DoubleAttr(override val value: Double) : AttributeValue
